spec/main_io_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'spec_helper'
+require 'main_io'
+
+class FakeStdin
```

Comment:
With these fakes defined with static methods + static attribute, does this mean that, for example, if you set the string in one test and get the string in a subsequent test, the string will be what you set it in the prior test?

If so, I'd term that "shared state" across tests, which can bite you pretty bad. Ideally each test cannot be affected by what happens in another test. If this isn't the case, it can get pretty ugly trying to figure out why a test is failing. You could even have tests fail or not fail depending on the ordering of tests. An easy fix in this case is make these normal methods and new-up instances of these fakes where you need to use them.

Reply:
Thanks for the suggestion. I implemented it!